pkg/apiserver/domain/service/pipeline.go

```diff
@@ -110,43 +124,92 @@ func NewContextService() ContextService {
 
 // CreatePipeline will create a pipeline
 func (p pipelineServiceImpl) CreatePipeline(ctx context.Context, req apis.CreatePipelineRequest) (*apis.PipelineBase, error) {
-	wf := v1alpha1.Workflow{}
-	wf.SetName(req.Name)
-	wf.SetNamespace(nsForProj(req.Project))
-	wf.WorkflowSpec = req.Spec
-	wf.SetLabels(map[string]string{
-		labelDescription: req.Description,
-		labelAlias:       req.Alias})
-	if err := p.KubeClient.Create(ctx, &wf); err != nil {
+	project := ctx.Value(&apis.CtxKeyProject).(*model.Project)
+	pipeline := &model.Pipeline{
+		Name:        req.Name,
+		Description: req.Description,
+		Alias:       req.Alias,
+		Project:     project.Name,
+		Spec:        req.Spec,
```

Comment:
Check the `req.Spec`. Such as:

1. The steps can not  be empty.
2. The mode is DAG or StepByStep.